thirdparty/hairer/radau5_superlu_double.c

```diff
@@ -0,0 +1,221 @@
+#include "slu_mt_ddefs.h"
+#include "radau5_superlu_double.h"
+
+struct SuperLU_aux_d{
+    int nprocs, n, nnz_jac;
+    int setup_done, fact_done; // flags for which memory to free in the end
+
+    double *data_sys;
+    int *indices_sys, *indptr_sys;
+
+    int panel_size, relax; // System specific tuning parameters
+    fact_t fact; // if factorized matrix is being supplied, if not: how to factorize
+    trans_t trans; // whether to solve transposed system or not
+    yes_no_t refact; // NO for first time fac., YES for re-factorization
+    yes_no_t usepr; // Whether the pivoting will use perm_r specified by the user, NO = it becomes output of pdgstrf function
+
+    double diag_pivot_thresh, drop_tol;
+    int lwork;
+    void *work;
+
+    int *perm_r, *perm_c; // row & column permutation vectors
+    Gstat_t *Gstat;
+    superlumt_options_t *slu_options;
+
+    SuperMatrix *A, *B, *AC, *L, *U;
+};
+
+// Initialization of required data structures for SuperLU
+SuperLU_aux_d* superlu_init_d(int nprocs, int n, int nnz){
+    SuperLU_aux_d *slu_aux = (SuperLU_aux_d *)malloc(sizeof(SuperLU_aux_d));
+    if (!slu_aux) {SUPERLU_ABORT("Malloc failed for slu_aux.");}
+    slu_aux->setup_done = 0;
+    slu_aux->fact_done = 0;
+
+    slu_aux->nprocs = nprocs;
+    slu_aux->nnz_jac = nnz;
+    slu_aux->n = n;
+
+    slu_aux->panel_size = sp_ienv(1);
+    slu_aux->relax = sp_ienv(2);
+
+    slu_aux->A =  (SuperMatrix *)malloc(sizeof(SuperMatrix));
+    slu_aux->B =  (SuperMatrix *)malloc(sizeof(SuperMatrix));
+    slu_aux->AC = (SuperMatrix *)malloc(sizeof(SuperMatrix));
+    slu_aux->L =  (SuperMatrix *)malloc(sizeof(SuperMatrix));
+    slu_aux->U =  (SuperMatrix *)malloc(sizeof(SuperMatrix));
+
+    if (!slu_aux->A)  {SUPERLU_ABORT("Malloc failed for A.");}
+    if (!slu_aux->B)  {SUPERLU_ABORT("Malloc failed for B.");}
+    if (!slu_aux->AC) {SUPERLU_ABORT("Malloc failed for AC.");}
+    if (!slu_aux->L)  {SUPERLU_ABORT("Malloc failed for L.");}
+    if (!slu_aux->U)  {SUPERLU_ABORT("Malloc failed for U.");}
```

Comment:
The abort messages in this file are identical to the ones in radau5_superlu_complex, perhaps consider if we should make them distinguishable from each other. Not mandatory but just for consideration.

Reply:
Well spotted, changed these to 
"Malloc failed for doublecomplex A"
resp.
"Malloc failed for double A"

Also did the same for a few more instances.